### PR TITLE
refactor: unify logic for blocks removal

### PR DESCRIPTION
**Pull request by klkvr**
Builds on top of https://github.com/paradigmxyz/reth/pull/12741

Similarly to https://github.com/paradigmxyz/reth/pull/12694 this unifies code paths for block removal.

Bodies removal is handled in `remove_bodies_above` method which is called directly in `BodyStage`. During live sync the blocks removal is now handled through `remove_blocks_above` method which is similar to `remove_block_range` which is now removed.

`take_block_range` is also removed and its usage in `take_block_and_execution_range` is replaced with a call to `sealed_block_with_senders_range` followed by `remove_blocks_above`

Block removal methods now also operate on `StorageLocation` enum.
- `StorageLocation::Database` is used by legacy engine to make sure that we don't touch static files which might conflict with pruner task
- `StorageLocation::Both` is used by everything else

The two additional changes are:
- Unifying code paths for consistency checks in `BodyStage`
- Changing interface of some of the provider functions to just accept unwind target instead of a block range.

### CODE REVIEW

File: crates/cli/commands/src/stage/unwind.rs [hunk 1]
```diff
@@ -186,10 +180,10 @@ impl Subcommands {
     /// Returns the block range to unwind.
     ///
     /// This returns an inclusive range: [target..=latest]
```
Comment (mattsse) [inline crates/cli/commands/src/stage/unwind.rs:182]: needs docs update

File: crates/storage/provider/src/traits/block.rs [hunk 2]
```diff
@@ -34,15 +33,17 @@ impl StorageLocation {
 #[auto_impl::auto_impl(&, Arc, Box)]
 pub trait BlockExecutionWriter: BlockWriter + Send + Sync {
     /// Take range of blocks and its execution result
-    fn take_block_and_execution_range(
+    fn take_block_and_execution_range_above(
```
Comment (mattsse) [inline crates/storage/provider/src/traits/block.rs:36]: range_above doesn't make much sense to me

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/storage/provider/src/traits/block.rs (hunk 2 above)
Comment (klkvr) [reply crates/storage/provider/src/traits/block.rs:36]: updated

**Review by joshieDo — APPROVED** [review]

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**